polylogue/db.py

```
@@ -417,42 +504,84 @@ def replace_messages(
-            INSERT INTO messages (
-                provider, conversation_id, branch_id, message_id, parent_id,
-                position, timestamp, role, model, content_hash, content_text,
-                content_json, rendered_text, raw_json, token_count, word_count,
-                attachment_count, attachment_names, is_leaf, metadata_json
+        info_rows = conn.execute("PRAGMA table_info(messages)").fetchall()
+        cols: set[str] = set()
+        for row in info_rows:
+            try:
+                cols.add(row["name"])  # type: ignore[index]
+            except Exception:
+                cols.add(row[1])
```

Ruff (0.15.9)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)